### PR TITLE
78265 get projects

### PR DESCRIPTION
**Pull request by llykkelarsen**
endpoint to get all projects in given plant from main api
changes in MainApiProjectServiceTests got a little confusing in git, but all I've done is add 4 new tests

### CODE REVIEW

File: src/Equinor.ProCoSys.IPO.MainApi/CommPkg/ICommPkgApiService.cs [hunk 1]
```diff
@@ -5,6 +5,6 @@ namespace Equinor.ProCoSys.IPO.MainApi.CommPkg
 {
     public interface ICommPkgApiService
     {
-        Task<IList<ProcosysCommPkg>> SearchCommPkgsByCommPkgNoAsync(string plant, int projectId, string startsWithTagNo);
+        Task<IList<ProCoSysCommPkg>> SearchCommPkgsByCommPkgNoAsync(string plant, int projectId, string startsWithTagNo);
```
Comment (bratli) [inline src/Equinor.ProCoSys.IPO.MainApi/CommPkg/ICommPkgApiService.cs:8]: Isnt it starsWithCommPkgNo?

File: src/Equinor.ProCoSys.IPO.MainApi/Project/MainApiProjectService.cs [hunk 2]
```diff
@@ -38,5 +41,28 @@ public async Task<ProCoSysProject> TryGetProjectAsync(string plant, string name)
...
         }
+
+        public async Task<IList<ProCoSysProject>> GetProjectsInPlantAsync(string plant)
+        {
+            if (!await _plantCache.IsValidPlantForCurrentUserAsync(plant))
+            {
+                throw new ArgumentException($"Invalid plant: {plant}");
+            }
+
+            var items = new List<ProCoSysProject>();
```
Comment (bratli) [inline src/Equinor.ProCoSys.IPO.MainApi/Project/MainApiProjectService.cs:52]: rename to projects just for better description of variable name

File: src/Equinor.ProCoSys.IPO.MainApi/CommPkg/ICommPkgApiService.cs (hunk 1 above)
Comment (llykkelarsen) [reply src/Equinor.ProCoSys.IPO.MainApi/CommPkg/ICommPkgApiService.cs:8]: oops yes just a forgotten commit

**Review by bratli — APPROVED** [review]
Just super picky on the variable name, not super important

**Pull request merged**